pkg/server/server.go

```diff
@@ -110,6 +132,42 @@ func (s *Server) shutdown() {
 	return
 }
 
+func (s *Server) setupProfiling() {
+	if s.Config.ProfilingEnabled {
+		if runtime.MemProfileRate == 0 {
+			s.Config.Log.Warn("Memory profiles are disabled")
```

Comment:
I'm curious under what conditions this can occur?

Reply:
Taken from https://golang.org/pkg/runtime/:
```
memprofilerate: setting memprofilerate=X will update the value of runtime.MemProfileRate.
When set to 0 memory profiling is disabled.  Refer to the description of
MemProfileRate for the default value.
```
So this can occur when GODEBUG variable has memprofilerarte=0 set.